src/prefect/tasks.py

```diff
@@ -387,8 +387,23 @@ def __init__(
         self.cache_expiration = cache_expiration
         self.refresh_cache = refresh_cache
 
+        # result persistence settings
         if persist_result is None:
-            persist_result = PREFECT_RESULTS_PERSIST_BY_DEFAULT.value()
+            if cache_policy and cache_policy != NONE and cache_policy != NotSet:
+                persist_result = True
+            elif cache_key_fn is not None:
+                persist_result = True
+            elif any(
+                [
+                    result_storage_key is not None,
+                    result_storage is not None,
+                    result_serializer is not None,
+                ]
+            ):
+                persist_result = True
```

Comment:
Seems like we could consolidate this all into one check?
```suggestion
            if any(
                [
                    cache_policy and cache_policy != NONE and cache_policy != NotSet,
                    cache_key_fn is not None,
                    result_storage_key is not None,
                    result_storage is not None,
                    result_serializer is not None,
                ]
            ):
                persist_result = True
```